File: 프로그래머스/1/12910. 나누어 떨어지는 숫자 배열/나누어 떨어지는 숫자 배열.js
function solution(arr, divisor) {
    let answer = [];
    
    for(const s of arr){
        
        if(s % divisor === 0){
            answer.push(s);
        }
    }
    
    if(answer.length === 0){
        answer.push(-1);    
    }
    
    
    return answer.sort(function(a,b) {return a-b});
}

File: 프로그래머스/1/12910. 나누어 떨어지는 숫자 배열/나누어 떨어지는 숫자 배열.java
import java.util.*;

class Solution {
    public int[] solution(int[] arr, int divisor) {
        
        ArrayList<Integer> list = new ArrayList<>();
        
        for(int i = 0; i < arr.length; i++){
            boolean check = (arr[i] % divisor == 0);
                
            if(check){
                list.add(arr[i]);
            }
        }
        
        if(list.size() == 0){
            return new int[] {-1};
        }
        
        int[] answer = new int[list.size()];
        
        for(int i = 0; i<list.size(); i++){
            answer[i] = list.get(i);
        }
        
        Arrays.sort(answer);
        
        return  answer;
    }
}